classes/class-p4-search.php

```diff
@@ -1020,6 +1034,59 @@ public static function edit_search_mime_types( $where ) : string {
 			return $where;
 		}
 
+		/**
+		 * Intercept ElasticPress request filter.
+		 * All these parameters are passed from EP.
+		 *
+		 * @param object $request Request object from ElasticPress.
+		 * @param array  $query Query from ElasticPress.
+		 * @param array  $args Arguments for the request.
+		 * @param array  $failures Errors array.
+		 */
+		public static function intercept_ep_request( $request, $query = null, $args = null, $failures = null ) {
+			$total_docs_count    = 0;
+			$indexed_docs_count  = 0;
+			$filtered_body_lines = [];
+
+			// Is this a bulk index task on the attachment pipeline?
+			if ( $query && ! ! preg_match( '/_bulk\?pipeline.*attachment/', $query['url'] ) ) {
```

Comment:
This filter would allow achieving the same thing, but in a much easier way: https://github.com/10up/ElasticPress/blob/develop/includes/classes/Indexable/Post/Post.php#L318

It contains the indexed item before it is passed to the bulk request or json encoded. If we check the mime type there then we can avoid it making it into the bulk request in the first place by returning `false`. That means we could achieve the same thing as here with about 4 lines of code, instead of 40, and no regex :)

Reply:
Hmmm but how would you actually prevent the "post" (for example, an image) from being sent to the index by using this filter? 

According to the blockdoc it seems like it's just there for backwards compatibility, it's being applied inside the Elasticpress code just to do a little transform in the structure of the arguments here:

https://github.com/10up/ElasticPress/blob/3b482e632521310dbb82d9602d54bf5f2b281c9c/includes/classes/Feature/Documents/Documents.php#L213

But there's no way to prevent the sync from actually happening, or I'm not seeing it 🤔